src/transformStateWithClones.js

```
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

This violates checklist item #3: "3. [CODE STYLE]: switch/case should always have default case for error handling." When you move to a `switch`, include a `default` branch that handles unknown `type` values (for example, throw an Error or otherwise handle the unexpected action). Right now unknown action types are silently ignored.